### PR TITLE
[ROCm] Fixed unexpected index error on handle convert on gemm rewriter

### PR DESCRIPTION
**Pull request by i-chaochen**
This https://github.com/openxla/xla/pull/2859/files#diff-44a446072b16823f5bb6016140330271e4a5a94ddc18c1d737439463213223fdR611-R630 caused some errors on our side, this PR is to fix it without awkward #if 

@akuegel  @ddunl Thanks in advance for reviewing it!

As a side note, any suggestion on our hipblas_lt PR?  https://github.com/tensorflow/tensorflow/pull/60894

### CODE REVIEW

File: xla/service/gpu/gemm_rewriter.cc [hunk 1]
```diff
@@ -608,13 +608,14 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
   Status HandleConvert(HloInstruction *instr) override {
     HloInstruction *clamp_lower, *clamp_upper, *d_scale, *existing_gemm,
         *binary;
-    auto cuda_compute_capability_ =
-        std::get<se::CudaComputeCapability>(gpu_version_);
+    bool gpu_capability_flag_ = (std::holds_alternative<se::CudaComputeCapability>(gpu_version_) &&
```
Comment (cheshire) [inline xla/service/gpu/gemm_rewriter.cc:611]: Maybe use a better var name?

Comment (i-chaochen) [reply xla/service/gpu/gemm_rewriter.cc:611]: @cheshire Thanks for the feedback! 

May I ask could you be more specific or any suggestion?  is ```gpu_cc_convert_flag_``` ok?

Comment (cheshire) [reply xla/service/gpu/gemm_rewriter.cc:611]: supports_different_output_type?

Comment (i-chaochen) [reply xla/service/gpu/gemm_rewriter.cc]: Done.

**Review by cheshire — APPROVED** [review]

**Pull request closed without merging**